common/buildings/hfe_buildings.txt

```diff
@@ -152,16 +167,79 @@ hfe_building_terraformer_hive_mind = {
 
 hfe_building_virtual = {
 	icon = building_virtual_entertainment_studios
+	base_cap_amount = 1
 	base_buildtime = @fetime
-	can_build = no
-	#can_demolish = no
+	can_demolish = no
+	can_be_ruined = no
+	can_be_disabled = yes
 	category = amenity
+	triggered_desc = {
+		text = VIRTUAL_FUTURE
+	}
+	potential = {
+		OR = {
+			owner = { has_country_flag = hfe_future }
+			planet = { has_planet_flag = hfe_future_capital }
+			planet = { has_planet_flag = hfe_future_planet }
+			planet = { has_building = hfe_building_virtual } # Once it's built, it stays
+		}
+	}
+	planet_modifier = {
+		planet_amenities_add = 50
+	}
 	resources = {
 		category = planet_buildings
+		cost = {
+			influence = 150
+			energy = 10000
+			alloys = 1000
+			minerals = 1000
+			rare_crystals = 750
+			exotic_gases = 100
+			volatile_motes = 5
+		}
+		produces = {
+			society_research = 30
+			consumer_goods = 30
+		}
 		upkeep = {
-			energy = 4
+			energy = 20
+			engineering_research = 20
+		}
+	}
+	ai_weight = {
+		weight = 10
+		modifier = {
+			factor = 0
+			owner = {
+				has_monthly_income = {
+					resource = energy
+					value < 30
+				}
+			}
+		}
+		modifier = {
+			factor = 0
+			owner = {
+				has_monthly_income = {
+					resource = engineering_research
+					value < 250
+				}
+			}
 		}
 	}
+	ai_resource_production = {
```

Comment:
I have no idea what is ai_resource_production

Reply:
AI Resource Production is the code that allows Fallen Empire Special Resource Engines to give a FE their Dark Matter without giving it to the player. It is used to keep the AI alive without giving what is essentially OP values to the player. Please don't touch that, I've worked on that balance for over a month and it is ESSENTIAL.